itests/hive-unit/src/test/java/org/apache/hive/beeline/TestHplSqlViaBeeLine.java

```diff
@@ -1411,8 +1404,174 @@ private void testScriptFile(String scriptText, List<String> argList, String expe
     testScriptFile(scriptText, argList, expectedPattern, OutStream.OUT);
   }
 
+  /**
+   * Checks that the pattern appears in the output.
+   * <p><b>Warning:</b> If the output depends on the time of the execution,
+   * use {@link #testTimeScriptFile}. This may be the case if
+   * the script contains CURRENT_DATE, CURRENT_TIME_MILLIS, etc.</p>
+   */
   private void testScriptFile(String scriptText, List<String> argList, String expectedPattern,
           TestBeeLineWithArgs.OutStream outStream) throws Throwable {
+    String output = runScriptFile(scriptText, argList, outStream);
+    if (!Pattern.compile(".*" + expectedPattern + ".*", Pattern.DOTALL).matcher(output).matches()) {
+      fail("Output: '" + output + "' should match " + expectedPattern);
+    }
+  }
+
+  interface TimeCheckHelper {
```

Comment:
If we drop the `withPrefixPattern` method we could turn this type-safe enum pattern (via `interface`) to an actual `enum`.